ad-hoc-scripts/updateEstablishmentsWithoutAPrimaryUser.sql

```diff
@@ -0,0 +1,169 @@
+                        ) IN (
+                                SELECT
+                                        u."EstablishmentID",
+                                        MIN(u."RegistrationID") userRegistrationID
+                                FROM
+                                        cqc."User" u
+                                        JOIN cqc."Login" l ON l."RegistrationID" = u."RegistrationID"
+                                WHERE
+                                        u."IsPrimary" = false
+                                        AND (
+                                                SELECT
+                                                        count(*)
+                                                FROM
+                                                        cqc."User"
+                                                WHERE
+                                                        "IsPrimary" = true
+                                                        AND "EstablishmentID" = u."EstablishmentID"
+                                        ) = 0
+                                        AND u."EstablishmentID" IS NOT NULL
+                                        AND u."UserRoleValue" = 'Edit'
+                                GROUP BY
+                                        (u."EstablishmentID")
+                        )
+        );
+
+-- Repeat count - should equal to zero
+SELECT
+        COUNT(*)
+FROM
+        (
+                SELECT
+                        DISTINCT ON (u."EstablishmentID") u."EstablishmentID"
+                FROM
+                        cqc."User" u
+                        JOIN cqc."Login" l ON l."RegistrationID" = u."RegistrationID"
+                WHERE
+                        u."IsPrimary" = false
+                        AND (
+                                SELECT
+                                        count(*)
+                                FROM
+                                        cqc."User"
+                                WHERE
+                                        "IsPrimary" = true
+                                        AND "EstablishmentID" = u."EstablishmentID"
+                        ) = 0
+                        AND u."EstablishmentID" IS NOT NULL
+                        AND u."UserRoleValue" = 'Edit'
+        ) as accountsWithoutPrimary;
+
+SELECT
+        DISTINCT ON (u."EstablishmentID") u."EstablishmentID",
+        l."Username"
+FROM
+        cqc."User" u
+        JOIN cqc."Login" l ON l."RegistrationID" = u."RegistrationID"
+WHERE
+        u."IsPrimary" = false
+        AND (
+                SELECT
+                        count(*)
+                FROM
+                        cqc."User"
+                WHERE
+                        "IsPrimary" = true
+                        AND "EstablishmentID" = u."EstablishmentID"
+        ) = 0
+        AND u."EstablishmentID" IS NOT NULL
+        AND u."UserRoleValue" = 'Edit';
+
+ROLLBACK;
```

Comment:
I ran this locally and got

```
WARNING:  there is no transaction in progress

Query 1 OK: ROLLBACK
```

Think this is doing the queries and then rolling back. Is that what you want?

Would need to be `END;` in order to commit the changes.

Reply:
Yeah, we want to commit the changes. We just opened the PR initially with the `ROLLBACK` to get eyes on it. 

I'll update it now